src/OrbitBase/include/OrbitBase/TemporaryFile.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TEMPORARY_FILE_H_
+#define ORBIT_BASE_TEMPORARY_FILE_H_
+
+#include "OrbitBase/File.h"
+
+namespace orbit_base {
+class TemporaryFile final {
+ public:
+  TemporaryFile() { Create(); }
+  ~TemporaryFile() { CloseAndRemove(); }
```

Comment:
I feel like having both the destructor and a public `CloseAndRemove` is a bit confusing. Can't the class just be purely RAII? This makes for a mix of styles and a not-so-clean interface.